aws/ci_build.rake

```diff
@@ -203,16 +197,18 @@ def upgrade_frontend(name, host)
     'git pull --ff-only',
     'sudo bundle install',
     'rake build',
+    # Use the AWS-provided script to cleanly re-register a frontend instance with its load balancer(s).
+    'bin/deploy_frontend/register_with_elb.sh'
   ]
   command = commands.join(' && ')
 
   HipChat.log "Upgrading <b>#{name}</b> (#{host})..."
 
   log_path = aws_dir "deploy-#{name}.log"
 
-  # Stop the frontend before running the commands so that the git pull doesn't modify files
-  # out from under a running instance. The rake build command will restart the instance.
-  stop_frontend name, host, log_path
+  # Remove the frontend from load balancer rotation before running the commands,
+  # so that the git pull doesn't modify files out from under a running instance.
+  deregister_frontend name, host, log_path
 
   success = false
   begin
```

Comment:
this line was commented by 65abdb7abe0cd6854082ff33ef4df77c58858825 on 12/7/2014 - the commit itself is opaque, but my guess would that it was done to reduce the amount of chat-room spam when deploying updates across our fleet of servers during Hour of Code that year.